wandb/sdk/artifacts/artifact.py

```diff
@@ -1466,26 +1477,20 @@ def add(self, obj: WBValue, name: StrPath) -> ArtifactManifestEntry:
         val = obj.to_json(self)
         name = obj.with_suffix(name)
         entry = self.manifest.get_entry_by_path(name)
-        if entry is not None:
+        if (overwrite is False) and (entry is not None):
```

Comment:
Made me scroll up to check if overwrite is `bool | None` :) would the following be better?

```suggestion
        if not overwrite and entry is not None:
```

(parentheses OK too)